### PR TITLE
Announcement management APIs

### PR DESCRIPTION
**Pull request by devformatters**

### CODE REVIEW

**Review by jeremyOk — APPROVED** [review]

File: src/types/system.ts [hunk 1]
```diff
@@ -224,6 +242,25 @@ export default /* GraphQL */ `
...
+  input AnnouncementsInput {
+    id: ID
+  }
+
+  input PutAnnouncementInput {
+    id: ID
+    title: String
+    cover: String
+    content: String
+    link: String
```
Comment (robertu7) [inline src/types/system.ts:254]: use `@constraint`?

File: src/types/system.ts [hunk 2]
```diff
@@ -43,17 +45,33 @@ export default /* GraphQL */ `
...
 
+  type Announcement {
+    id: ID!
+    title: String
+    cover: String
+    content: String
+    link: String
+    type: AnnouncementType!
+    visible: Boolean!
+    order: Int!
```
Comment (robertu7) [inline src/types/system.ts:70]: where's the `order` uses for? looks `official/announcements` don't order by `order`.

**Review by robertu7 — APPROVED** [review]

File: src/types/system.ts (hunk 1 above)
Comment (devformatters) [reply src/types/system.ts]: Fixed. Thanks.

File: src/types/system.ts (hunk 2 above)
Comment (devformatters) [reply src/types/system.ts:70]: It's a optional field for future implementation.

**Pull request merged**